### PR TITLE
CASMNET-2127 | qlogic driver crash known issue doc

### PR DESCRIPTION
**Pull request by kosonenj**
# Description

Known  issue documentation for Qlogic driver crash triaged in CASMTRIAGE-5033

Relates to:
- https://jira-pro.it.hpe.com:8443/browse/CASMNET-2127
- https://jira-pro.it.hpe.com:8443/browse/CASMTRIAGE-5033

# Checklist

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+In some failover/maintenance scenarios users may experience Qlogic driver crash on the storage nodes. This issue is usually tied to network event's such as an reboot of a switch in the VSX pair but can happen with out user intervention as well albeit rarely.
+
+Issue seems to be aggravated by incorrect configuration for storage node back-channel links and per internal testing correcting the configuration on switch greatly reduces the risk of running into this issue.
+
+The configuration is fixed in Canu versions 2.3 and above.
+
+Signature of the driver crash:
+
+```bash
```
Comment (rustydb) [inline troubleshooting/known_issues/qlogic_driver_crash.md:13]:
```suggestion
```text
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+## Workaround
+
+Upgrade Canu to 2.3 or above, regenerate the configuration and apply it the VSX-pair that hosts the storage nodes.
+
+Or manually correct the configuration.
+
+In this example of the problem you can see that the lags 10,12,14 which are the lags for the storage back channel communication have applied configuration of "vlan trunk native 1". Which is wrong.
+
+```bash
```
Comment (rustydb) [inline troubleshooting/known_issues/qlogic_driver_crash.md:58]:
```suggestion
```text
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    vlan trunk native 1
+    vlan trunk allowed 10
+    lacp mode active
+    lacp fallback
+    spanning-tree port-type admin-edge
+```
+
+Too correct the issue, please login to both the VSX pair switches and correct the configuration of these lags to use "vlan trunk native 10":
+
+```bash
```
Comment (rustydb) [inline troubleshooting/known_issues/qlogic_driver_crash.md:90]:
```suggestion
```text
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    no shutdown
+    no routing
+    vlan trunk native 1
+    vlan trunk allowed 10
+    lacp mode active
+    lacp fallback
+    spanning-tree port-type admin-edge
+```
+
+Too correct the issue, please login to both the VSX pair switches and correct the configuration of these lags to use "vlan trunk native 10":
```
Comment (rustydb) [inline troubleshooting/known_issues/qlogic_driver_crash.md:88]:
```suggestion
Too correct the issue, please login to both the VSX pair switches and correct the configuration of these lags to use `vlan trunk native 10`:
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+[72201.695800] RIP: 0033:0x7f4a2d503a3e 
+```
+
+## Workaround
+
+Upgrade Canu to 2.3 or above, regenerate the configuration and apply it the VSX-pair that hosts the storage nodes.
+
+Or manually correct the configuration.
+
+In this example of the problem you can see that the lags 10,12,14 which are the lags for the storage back channel communication have applied configuration of "vlan trunk native 1". Which is wrong.
```
Comment (rustydb) [inline troubleshooting/known_issues/qlogic_driver_crash.md:56]:
```suggestion
In this example of the problem you can see that the lags 10,12,14 which are the lags for the storage back channel communication have applied configuration of `vlan trunk native 1`. Which is wrong.
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
+# Qlogic driver crash
+
+## Description
+
+In some failover/maintenance scenarios users may experience Qlogic driver crash on the storage nodes. This issue is usually tied to network event's such as an reboot of a switch in the VSX pair but can happen with out user intervention as well albeit rarely.
```
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:5]:
```suggestion
In some failover/maintenance scenarios users may experience Qlogic driver crash on the storage nodes. This issue is usually tied to network events, such as a reboot of a switch in the VSX pair, but can also occur without user intervention in rare events.
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+Issue seems to be aggravated by incorrect configuration for storage node back-channel links and per internal testing correcting the configuration on switch greatly reduces the risk of running into this issue.
+
+The configuration is fixed in Canu versions 2.3 and above.
```
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:9]:
```suggestion
The configuration is fixed in CANU versions 2.3 and above.
```
CANU is all uppercase throughout the rest of the documentation.

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+[72201.674690]  ? do_syscall_64+0x67/0x80
+[72201.679335]  ? do_syscall_64+0x67/0x80
+[72201.683978]  ? exit_to_user_mode_prepare+0x1dc/0x230
+[72201.689845]  entry_SYSCALL_64_after_hwframe+0x61/0xcb
+[72201.695800] RIP: 0033:0x7f4a2d503a3e 
+```
+
+## Workaround
+
+Upgrade Canu to 2.3 or above, regenerate the configuration and apply it the VSX-pair that hosts the storage nodes.
```
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:52]:
```suggestion
Upgrade CANU to version 2.3 or above, regenerate the configuration, and then apply it the VSX-pair that hosts the storage nodes.
```

File: troubleshooting/known_issues/qlogic_driver_crash.md (hunk 4 above)
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:88]:
```suggestion
To correct the issue, log in to both the VSX pair switches and correct the configuration of these lags to use "vlan trunk native 10":
```

File: troubleshooting/known_issues/qlogic_driver_crash.md (hunk 5 above)
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:56]:
```suggestion
In the following example, lags 10, 12, and 14, which are the lags for the storage back channel communication, have a configuration of "vlan trunk native 1". This is incorrect.
```

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    vlan trunk native 10
+    vlan trunk allowed 10
+    lacp mode active
+    lacp fallback
+    spanning-tree port-type admin-edge
+```
+
+## Fix
+
+Working with vendor to fix qlogic driver code, once fix is released we will add the new qlogic driver to our firmware pack.
```
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:122]:
```suggestion
The vendor is working to fix the Qlogic driver code. Once a fix is released, the new Qlogic driver will be added to the firmware pack.
```

File: .spelling [hunk 10]
```diff
@@ -885,6 +885,11 @@ twinax
 USR SFP-10G
 vcs-user-credentials
 Xcvr
+Qlogic
+qlogic
```
Comment (rustydb) [inline .spelling:889]:
```suggestion
QLogic
qlogic
```

nit: when using the formal name, it should be "QLogic."

**Review by rustydb — CHANGES_REQUESTED** [review]

File: troubleshooting/known_issues/qlogic_driver_crash.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    no shutdown
+    no routing
+    vlan trunk native 1
+    vlan trunk allowed 10
+    lacp mode active
+    lacp fallback
+    spanning-tree port-type admin-edge
+```
+
+Too correct the issue, please login to both the VSX pair switches and correct the configuration of these lags to use `vlan trunk native 10`:
```
Comment (nrockershousen) [inline troubleshooting/known_issues/qlogic_driver_crash.md:88]:
```suggestion
To correct the issue, log in to both the VSX pair switches and correct the configuration of these lags to use `vlan trunk native 10`:
```

**Review by nrockershousen — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**